routing/pathfind.go

```diff
@@ -90,17 +90,26 @@ type edgePolicyWithSource struct {
 	edge       *channeldb.ChannelEdgePolicy
 }
 
+// finalHopParams encapsulates various parameters for route construction that
+// apply to the final hop in a route. These features include basic payment data
+// such as amounts and cltvs, as well as more complex features like destination
+// TLV records.
+type finalHopParams struct {
+	amt       lnwire.MilliSatoshi
+	cltvDelta uint16
+	records   []tlv.Record
```

Comment:
I am converting this into `hop.CustomRecordSet` in https://github.com/lightningnetwork/lnd/pull/3819. But will rebase if this pr goes first.

Reply:
ideally we'd get #3819 in first, then i can rebase over it